### PR TITLE
[Merged by Bors] - feat: add some lemmas about polynomials and ofNat to enable simplifying with nat literals

### PR DESCRIPTION
**Pull request by alexjbest**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Polynomial/Eval.lean [hunk 1]
```diff
@@ -345,6 +345,16 @@ theorem eval₂_at_nat_cast {S : Type*} [Semiring S] (f : R →+* S) (n : ℕ) :
...
 
+@[simp]
+theorem eval₂_at_ofNat {S : Type*} [Semiring S] (f : R →+* S) (n : ℕ) [n.AtLeastTwo] :
+    p.eval₂ f (no_index (OfNat.ofNat n)) = f (p.eval (OfNat.ofNat n)) := by
+  simp [OfNat.ofNat]
+
+@[simp]
+lemma eval₂_ofNat {S : Type*} [Semiring S] (n : ℕ) [n.AtLeastTwo] (f : R →+* S) (a : S) :
+    (no_index (OfNat.ofNat n : R[X])).eval₂ f a = OfNat.ofNat n := by
+  simp [OfNat.ofNat]
```
Comment (eric-wieser) [inline Mathlib/Data/Polynomial/Eval.lean:356]: Can you add the `nat_cast` version of this too?

File: Mathlib/Data/Polynomial/Eval.lean [hunk 2]
```diff
@@ -354,6 +364,11 @@ theorem eval_C : (C a).eval x = a :=
 theorem eval_nat_cast {n : ℕ} : (n : R[X]).eval x = n := by simp only [← C_eq_nat_cast, eval_C]
 #align polynomial.eval_nat_cast Polynomial.eval_nat_cast
 
+@[simp]
+lemma eval_ofNat (n : ℕ) [n.AtLeastTwo] (a : R) :
+    (no_index (OfNat.ofNat n : R[X])).eval a = OfNat.ofNat n := by
+  simp [OfNat.ofNat]
```
Comment (eric-wieser) [inline Mathlib/Data/Polynomial/Eval.lean:370]: It would be nice to squeeze this simp for symmetry with the one above.

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

File: Mathlib/Data/Polynomial/Eval.lean (hunk 1 above)
Comment (alexjbest) [reply Mathlib/Data/Polynomial/Eval.lean:356]: that already exists above

Comment (eric-wieser) [reply Mathlib/Data/Polynomial/Eval.lean:356]: In that case can you move this lemma to be right below the `nat_cast` version?

Comment (alexjbest) [reply Mathlib/Data/Polynomial/Eval.lean:356]: sure...

**Pull request closed without merging**